File: Leetcode 0257 Binary Tree Paths [easy] [tree]/Binary Tree Paths [divide-conquer].java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<String> binaryTreePaths(TreeNode root) {
        List<String> paths = new ArrayList<>();
        
        if (root == null) {
            return paths;
        }
        
        if (root.left == null && root.right == null) {
            paths.add(root.val + "");
            return paths;
        }
        
        List<String> leftPaths = binaryTreePaths(root.left);
        List<String> rightPaths = binaryTreePaths(root.right);
        
        for (String path : leftPaths) {
            paths.add(root.val + "->" + path);
        }
        
        for (String path : rightPaths) {
            paths.add(root.val + "->" + path);
        }
        
        return paths;
    }
}

File: Leetcode 0257 Binary Tree Paths [easy] [tree]/Binary Tree Paths [divide-conquer].js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
var binaryTreePaths = function(root) {
    let paths = [];
    if (root === null) {
        return paths;
    }
    
    if (root.left === null && root.right === null) {
        paths.push(root.val + "");
    }
    
    let leftPaths = binaryTreePaths(root.left);
    let rightPaths = binaryTreePaths(root.right);
    
    for (let path of leftPaths) {
        paths.push(root.val + "->" + path);
    }
    
    for (let path of rightPaths) {
        paths.push(root.val + "->" + path);
    }
    
    return paths;
};